packages/layout/src/tree-layout/layout-definition/helper/index.js

```diff
@@ -0,0 +1,290 @@
+        });
+    } else if (node.model().host() && node.model().host().getLogicalSpace) {
+        const containerWidth = node.boundBox().width;
+        const hostWidth = node.model().host().getLogicalSpace().width;
+        smallestWidth = containerWidth - hostWidth;
+        if (smallestWidth < 0) {
+            smallestWidth = 0;
+        }
+    } else {
+        smallestWidth = 0;
+    }
+    return smallestWidth;
+}
+
+export function smallestExtraWidthVertically (node) {
+    let smallestWidth = Number.MAX_SAFE_INTEGER;
+    node.children().forEach((child) => {
+        const w = xExtraSpace(child);
+        if (w < smallestWidth) {
+            smallestWidth = w;
+        }
+    });
+    return smallestWidth;
+}
+
+export function determineBoundBox (bb, i, arr, instance) {
+    if (i) {
+    // if not first sibling, take boundbox from previous sibling
+        const lastSibling = arr[i - 1];
+        return {
+            width: bb.width,
+            height: bb.height,
+
+            top: instance.parentCut() === 'h'
+        ? lastSibling.boundBox().top + lastSibling.boundBox().height : lastSibling.boundBox().top,
+
+            left: instance.parentCut() === 'h'
+        ? lastSibling.boundBox().left : lastSibling.boundBox().left + lastSibling.boundBox().width
+        };
+    }
+  // if first sibling, take boundbox from parent
+    return {
+        width: bb.width,
+        height: bb.height,
+        top: instance.parent().boundBox().top,
+        left: instance.parent().boundBox().left
+    };
+}
+
+// prepares the targetComponent Map for target Mapping ie. where a component should lie
+export function prepareTargetComponentMap (context) {
+    context.targetComponentMap(new Map());
+    context.componentMap().forEach((value) => {
+        if (context.targetComponentMap().has(value.target())) {
+            context.targetComponentMap().get(value.target()).push(value);
+        } else {
+            const temp = [];
+            temp.push(value);
+            context.targetComponentMap().set(value.target(), temp);
+        }
+    });
+}
+
+export function getComponent (canvasComponent, componentName) {
+    const comp = canvasComponent.find(component => component.name() === componentName);
+    return (comp && comp !== -1) ? comp : null;
+}
+export function createPlaceHolderComponent (height, width, position) {
+    const comp = new DummyComponent(0, { height, width });
+    comp.name('placeHolder');
+    comp.position(position);
+    return comp;
+}
+
+export function placeComponent (definitionModel, component, isPreferred = false, isGridComponent = false) {
+    if (!component) {
+        return { first: definitionModel, second: definitionModel };
+    }
+    const componentDimension = component.getLogicalSpace();
+    const componentHeight = componentDimension.height;
+    const componentWidth = componentDimension.width;
+    let cut = '';
+    let componentRatioWidth = 1;
+    let leftOvercomponentRationWidth = 1;
```

Comment:
Define let at the beginning of the function, define const after that. Give one line space after declaration and logic